monitor/monitor.go

```diff
@@ -221,6 +216,34 @@ func (m *EthMonitor) monitorSlash() {
 	})
 }
 
+func (m *EthMonitor) monitorSubmitChange() {
+	m.monitorTendermintEvent(submitChangeEvent, func(e abci.Event) {
+		if !m.isValidator && !viper.GetBool(common.FlagSgnBootNode) {
+			return
+		}
+
+		event := sdk.StringifyEvent(e)
+		if event.Type == sync.EventTypeSync && event.Attributes[0].Value == sync.ActionSubmitChange {
+			changeId, err := strconv.ParseUint(event.Attributes[1].Value, 10, 64)
+			if err != nil {
+				log.Errorln("Parse changeId error", err)
+				return
+			}
+
+			change, err := sync.CLIQueryChange(m.operator.CliCtx, sync.RouterKey, changeId)
+			if err != nil {
+				log.Errorln("Query change error", err)
+				return
+			}
+
+			if m.verifyChange(change) {
+				msg := sync.NewMsgApprove(changeId, m.operator.Key.GetAddress())
```

Comment:
should this msg be signed? otherwise, a malicious node can modify code and use others address

Reply:
It is not necessary as discussed in person